File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s) {
    const stack=[];
    
    for(const char of s){
        if(char===')'){
            if(stack[stack.length-1]==='(') stack.pop();
            else return false;
        }
        else stack.push(char);
    }
    return stack.length>0 ? false: true;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;
class Solution {
    boolean solution(String s) {
        Stack<Character> stack=new Stack<>();
        // String[] arr = s.split("");
        
        for(int i=0; i<s.length(); i++){
            if(s.charAt(i)==')'){
                if(!stack.isEmpty()&&stack.peek()=='(') stack.pop();
                else stack.push(s.charAt(i));
            }
            else stack.push(s.charAt(i));
        }

        return stack.size()>0? false: true;
    }
}